templates/tftmpl/variables.go

```diff
@@ -57,26 +57,24 @@ func newVariablesTF(w io.Writer, filename string, input *RootModuleInputData) er
 		return err
 	}
 
+	// service variable is required to append
 	if _, err = w.Write(VariableServices); err != nil {
 		return err
 	}
 
-	if input.Condition != nil && input.Condition.SourceIncludesVariable() {
-		if err = input.Condition.appendVariable(w); err != nil {
-			return err
-		}
-	}
+	// append a variable for each monitored object
+	// note: assumes monitored objects all have a unique type. otherwise would
+	// need to check to avoid appending duplicate variables
+	for _, monitor := range input.Monitors {
+		if monitor.SourceIncludesVariable() {
+			if monitor.ServicesAppended() {
+				// services variable is already appended earlier. skip
+				continue
```

Comment:
Just want to make sure I understand this properly, this loop isn't just for the `services` variable, but if the monitor does happen to have `ServicesAppended` set to true, that means the monitor cares about `services` and no further action needs to be taken since we've already handled `services`. Is that correct?

Reply:
Yes, your understanding is correct! I think the name ServicesAppended() is a little confusing - ServicesAppended() will only returns true for MonitorServices (and never for the others). I will update the comment for the loop to clarify the `services` variable exception. Maybe I can also rename `ServicesAppended` to something like `IsServicesVar` to help clarify? Let me know if you have thoughts!

Reply:
Ooh, I think `IsServicesVar` is a good rename, that helps a lot!